pkg/tracer/tracer_linux.go

```diff
@@ -26,21 +29,47 @@ type LinuxTracer struct {
 	// In flight connections are the connections that already existed before the EBPF module was loaded.
 	// These connections are stored with a key without direction, to make it possible to merge with undirected
 	// metric stats
-	inFlightTCP map[string]*common.ConnectionStats
+	inFlightTCP     map[string]*common.ConnectionStats
+	metricsGatherer prometheus.Gatherer
+
+	perfEventsBytes   chan []byte
+	perfEventsLostLog chan uint64
+	perfMap           *bpflib.PerfMap
+
+	httpStatusCodes *prometheus.HistogramVec
+	stopCh          chan bool
 }
 
 var (
 	DebugFsPath  = "/sys/kernel/debug"
 	DebugFsMagic = int64(0x64626720) //http://man7.org/linux/man-pages/man2/statfs.2.html
+	PerfEventsBuffer = 100
 )
 
 func MakeTracer(config *config.Config) (Tracer, error) {
+
+	metricsGatherer := prometheus.NewRegistry()
+	httpStatusCodes := prometheus.NewHistogramVec(prometheus.HistogramOpts{
```

Comment:
Isn't there something in our own agent code that does the same thing so we don't need to add a dependency on prometheus. That seems a bit weird since our agent does very similar stuff already.